README.md

```diff
@@ -52,70 +92,117 @@ cloudscraper.get('http://website.com/', function(error, response, body) {
+ - Cloudscraper uses request-promise, promise chaining is done exactly the same as described in [docs](https://github.com/request/request-promise#cheat-sheet):
+ ```
+  cloudscraper(options)
+    .then(function (htmlString) {
+    })
+    .catch(function (err) {
+    });
+  ```
+
+## Default options
+Cloudscraper exposes following options that areq required by default but might be changed. Please note that default options increase chances of correct work.
+
 ```
+var options = {
+  uri: 'https://website',
+  jar: requestModule.jar(), // Custom cookie jar
+  headers: {
+    // User agent, Cache Control and Accept headers are required
+    'User-Agent': 'Ubuntu Chromium/34.0.1847.116 Chrome/34.0.1847.116 Safari/537.36',
+    'Cache-Control': 'private',
+    'Accept': 'application/xml,application/xhtml+xml,text/html;q=0.9, text/plain;q=0.8,image/png,*/*;q=0.5'
+  },
+  // Cloudflare requires a delay of 5 seconds, so wait for at least 6.
+  cloudflareTimeout: 6000,
+  // followAllRedirects - follow non-GET HTTP 3xx responses as redirects
+  followAllRedirects: true,
+  // Support only this max challenges in row. If CF returns more, throw an error
+  challengesToSolve: 3
+};
+
+cloudscraper(options, function(error, response, body) {
+  console.log(body)
+});
 
+```
+## Error object
+Cliudscraper error object inherits from `Error` has following fields:
+  * `name` - `RequestError`/`CaptchaError`/`CloudflareError`/`ParserError`
+  * `options` - The request options
+  * `cause` - An alias for `error`
+  * `response` - The request response
+  * `errorType` - Custom error code
 Where `errorType` can be following:
  - `0` if request to page failed due to some native reason as bad url, http connection or so. `error` in this case will be error [event](http://nodejs.org/api/http.html#http_class_http_server)
  - `1` cloudflare returned captcha. Nothing to do here. Bad luck
  - `2` cloudflare returned page with some inner error. `error` will be `Number` within this range `1012, 1011, 1002, 1000, 1004, 1010, 1006, 1007, 1008`. See more [here](https://support.cloudflare.com/hc/en-us/sections/200038216-CloudFlare-Error-Messages)
  - `3` this error is returned when library failed to parse and solve js challenge. `error` will be `String` with some details. :warning: :warning: __Most likely it means that cloudflare have changed their js challenge.__
  - `4` CF went into a loop and started to return challenge after challenge. If number of solved challenges is greater than `3` and another challenge is returned, throw an error
 
-
 Running tests
 ============
 Clone this repo, do `npm install` and then just `grunt`
 
 ### Unknown error? Library stopped working? ###
 Let me know, by opening [issue](https://github.com/codemanki/cloudscraper/issues) in this repo and i will update library asap. Please, provide url and body of page where cloudscraper failed.
 
-
-CloudScraper uses [Request](https://github.com/request/request) to perform requests.
```

Comment:
This is actually still true, `request-promise` only patches the init method and exports a freshly required version of `request`. If they didn't bypass the `require.cache` then `require('request-promise') === require('request')` would be true.

It might be better to keep that higher up in the README for those that are unfamiliar with `request-promise` but have used `request`.

Reply:
I removed it here, because it looks like a duplicate of `Dependencies` section which is at the bottom